wpilibc/src/main/native/include/frc/sysid/SysIdRoutineLog.h

```diff
@@ -61,87 +61,127 @@ class SysIdRoutineLog {
     /**
      * Log the voltage applied to the motor.
      *
+     * The value is recorded in volts by default;
+     * to change this, specify your target unit as a generic bound.
+     *
```

Comment:
```suggestion
     * @tparam U The unit to log in. Defaults to volts.
```
Specifying a target unit as a generic bound doesn't make sense- First, generic bound is a Java term, while in C++ the closest equivalent would be constraint. Second, a generic bound would be specified by the method, not something specified at the call site. Also, I think we require documentation for template parameters anyways, so there isn't as much of a need for a separate description.

Reply:
I think instead of using a generic it makes more sense to just change the unit in the function signature eg for lengths `units::length_unit auto` which should alllow passing in any length

Reply:
Maybe? Remember that the goal isn't to be able to pass in any type (which could be done already via implicit conversions), it's to output in any type. Logging in whatever unit is passed in is possible, but I'm not sure if that'd necessarily be what users expect.

Reply:
> Logging in whatever unit is passed in is possible, but I'm not sure if that'd necessarily be what users expect.

Thats what I'd expect. It can be documented as well.

> (which could be done already via implicit conversions)

I don't think this is true? See https://github.com/wpilibsuite/allwpilib/pull/6315 https://github.com/wpilibsuite/allwpilib/pull/6314

Reply:
> > Logging in whatever unit is passed in is possible, but I'm not sure if that'd necessarily be what users expect.
> 
> Thats what I'd expect. It can be documented as well.

If that's the direction we want to go, then sure. (Just remember to change Java to match.)

> > (which could be done already via implicit conversions)
> 
> I don't think this is true? See #6315 and #6314

Yes, there are already implicit conversions. #6314 failed because you can't chain two implicit user-defined conversions in a row- There was an implicit user-defined conversion from `units::degree_t` to `units::radian_t` and another from `units::radian_t` to `Rotation2d`, but since at most one implicit user-defined conversion is allowed (see https://en.cppreference.com/w/cpp/language/implicit_conversion), the build failed when trying to do an implicit conversion from degrees to `Rotation2d`. I'm not sure what #6315 is supposed to communicate.

Reply:
Instead of passing the desired unit as a template parameter, just have the user cast it themselves and use `MotorLog& voltage(units::voltage_unit auto voltage) {`.

Reply:
Should the same be done for Java?

Reply:
I'd say yes